src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,164 @@
+"""
+Defines a SCDataFrame class for use in coSMicQC.
+"""
+
+import pathlib
+from typing import Any, Dict, TypeVar, Union
+
+import pandas as pd
+
+# provide backwards compatibility for Self type in earlier Python versions.
+# see: https://peps.python.org/pep-0484/#annotating-instance-and-class-methods
+SCDataFrame_type = TypeVar("SCDataFrame_type", bound="SCDataFrame")
+
+
+class SCDataFrame:
+    """
+    A class to handle and load different types of data files into a pandas DataFrame.
+
+    This class can initialize with either a pandas DataFrame or a file path (CSV, TSV,
+    TXT, or Parquet). When initialized with a file path, it reads the data into a
+    pandas DataFrame.
+
+    Attributes:
+        data_source (str):
+            A string indicating the data source, either 'pd.DataFrame'
+            or the file path.
+        data (pd.DataFrame):
+            The loaded data in a pandas DataFrame.
+
+    Methods:
+        __call__():
+            Returns the underlying pandas DataFrame.
```

Comment:
Is this the default when using this class? Meaning it will return a pandas data frame when initialized unless calling another method?

Reply:
Thanks for this question - yes, this is the default `__call__` functionality of this class. I don't feel firmly that this will remain the case, but it felt for now a good way to build this class without coupling things together too tightly. The loose coupling of being able to use this class without needing to rely on it means we always have the Pandas DataFrame structures to rely on throughout while this class grows (or recedes) over time.